packages/teams-js/src/public/media.ts

```diff
@@ -412,30 +414,51 @@ export namespace media {
      * @param mediaEvent indicates what the event that needs to be signaled to the host client
      * Optional; @param callback is used to send app if host client has successfully handled the notification event or not
      */
-    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void {
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
       ensureInitialized(FrameContexts.content, FrameContexts.task);
-      const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
-      if (err) {
-        if (callback) {
-          callback(err);
-        }
-        return;
-      }
+      let mediaError;
+      const promise = Promise.resolve().then(() =>
+        throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion),
+      );
 
       const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+
       sendMessageToParent('media.controller', [params], (err?: SdkError) => {
         if (callback) {
-          callback(err);
+          return callback(err);
+        } else if (err) {
+          mediaError = err;
         }
       });
+
+      return promise
+        .then(() => {
+          if (mediaError) {
+            throw mediaError;
+          }
+        })
+        .catch(err => {
```

Comment:
nitpicky, but I don't think you need to catch and rethrow here, I believe if you just remove the catch the same error will be thrown to the caller